### PR TITLE
Fix Windows Cleanup Bug + Automatic GPU accelleration

### PR DESCRIPTION
**Pull request by Ma5onic**
Addresses the following issues: https://github.com/axeldelafosse/stemgen/issues/37 & https://github.com/axeldelafosse/stemgen/issues/9

**Feature Improvement for Issue https://github.com/axeldelafosse/stemgen/issues/9**:
- **Automatic GPU Detection**: Script now does a check to see if CUDA is available. If so, it will use the GPU by default (can be overridden by using the `-d`/`--device` flag).
If no CUDA support is detected, the script will use the CPU instead.

**Bug Fix for Issue https://github.com/axeldelafosse/stemgen/issues/37**:
- **Move out of the Target Directory to be cleaned**: Before attempting to remove the directory, I changed the current working directory to `OUTPUT_PATH` to ensure we are not inside the directory we're trying to delete. 
**Minor Improvements**
- **Path Handling**: Improved the way paths are constructed using `os.path.join()` consistently instead of a mix of string concatenation and joining. This ensures paths are constructed in a platform-independent manner.
- **Error Handling**: Added a `try-except` block around the `shutil.rmtree()` call. If a `PermissionError` occurs, a message is printed indicating the directory might still be in use. This provides a clearer error message and ensures the script doesn't crash due to this specific error.
- **Checking for Stem File**: Modified the path for checking and renaming the stem file to ensure it's constructed correctly.

### CODE REVIEW

File: stemgen.py [hunk 1]
```diff
@@ -47,13 +48,11 @@
     "-o",
     "--output",
     dest="OUTPUT_PATH",
-    default="output"
-    if str(INSTALL_DIR) == PROCESS_DIR or INSTALL_DIR.as_posix() == PROCESS_DIR
-    else ".",
+    default="output" if INSTALL_DIR.as_posix() == PROCESS_DIR else ".",
```
Comment (axeldelafosse) [inline stemgen.py:51]: Can you revert this change please? We need this additional condition on Windows

File: stemgen.py [hunk 2]
```diff
@@ -64,7 +63,15 @@
     else os.path.join(PROCESS_DIR, args.OUTPUT_PATH)
 )
 FORMAT = args.FORMAT
-DEVICE = args.DEVICE
+
+# Automatically set DEVICE to "cuda" if CUDA is available, otherwise set it to "cpu"
+DEVICE = args.DEVICE if args.DEVICE is not None else ("cuda" if torch.cuda.is_available() else "cpu")
```
Comment (axeldelafosse) [inline stemgen.py:68]: 💯

**Review by axeldelafosse — CHANGES_REQUESTED** [review]
Thank you for this PR! Just need a quick change before merging.

File: stemgen.py (hunk 2 above)
Comment (Ma5onic) [reply stemgen.py:68]: Yep! Sorry for the delay, got held up at work.

File: stemgen.py (hunk 1 above)
Comment (Ma5onic) [reply stemgen.py:51]: Yep! Sorry for the delay, got held up at work.

Comment (axeldelafosse) [reply stemgen.py:51]: > every variable would point to your current directory

Only when the `--output` argument is not specified. This is only affecting the default value.

**Review by axeldelafosse — APPROVED** [review]
